integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/lifecycle/plan/catalog/IcebergHandler.java

```diff
@@ -135,8 +136,7 @@ public DatasetIdentifier getDatasetIdentifier(
         maybeTableLocation.orElseGet(
             () -> reconstructDefaultLocation(new Path(warehouseLocation), identifier));
     DatasetIdentifier di = PathUtils.fromPath(tableLocation);
-    maybeSymlink.ifPresent(
-        symlink -> di.withSymlink(symlink.getName(), symlink.getNamespace(), SymlinkType.TABLE));
+    maybeSymlink.ifPresent(di::merge);
```

Comment:
`di` will contain dataset name and namespace extracted from table location and use it as a symlink. 
Why do you want to assign symlink type as `unknown`. Wouldn't  `TABLE_PATH` be a better choice? It looks like the code knows where the symlink comes from.

Reply:
In this concrete case, yes - although I would not name it `TABLE_PATH` - it's either a table or a path :) 

However, I plan to add more types that we miss - for example, we don't capture catalog name, and we probably should - under type `CATALOG`.

Reply:
so what's your opinion: stay with `unknown` or add other types?

Reply:
I left TABLE for now, however, will revisit that later.